packages/issuer/lib/VcIssuer.ts

```diff
@@ -0,0 +1,56 @@
+import { CredentialRequest, CredentialResponse, getKidFromJWT, IssuerMetadata, TokenErrorResponse } from '@sphereon/openid4vci-common'
+import { ICredential, W3CVerifiableCredential } from '@sphereon/ssi-types'
+
+export class VcIssuer {
+  _issuerMetadata: IssuerMetadata
+  _userPinRequired?: boolean
+  constructor(issuerMetadata: IssuerMetadata, userPinRequired?: boolean) {
+    this._issuerMetadata = issuerMetadata
+    this._userPinRequired = userPinRequired
+  }
+
+  public getIssuerMetadata() {
+    return this._issuerMetadata
+  }
+
+  public async issueCredentialFromIssueRequest(issueCredentialRequest: CredentialRequest): Promise<CredentialResponse> {
+    //TODO: do we want additional validations here?
+    if (this.isMetadataSupportCredentialRequestFormat(issueCredentialRequest.format)) {
+      return await this.issueCredential(issueCredentialRequest)
+    }
+    throw new Error(TokenErrorResponse.invalid_request)
+  }
+
+  private isMetadataSupportCredentialRequestFormat(requestFormat: string | string[]): boolean {
+    for (const credentialSupported of this._issuerMetadata.credentials_supported) {
+      if (!Array.isArray(requestFormat) && credentialSupported.format === requestFormat) {
+        return true
+      } else if (Array.isArray(requestFormat)) {
+        for (const format of requestFormat as string[]) {
+          if (credentialSupported.format === format) {
+            return true
+          }
+        }
+      }
+    }
+    return false
+  }
+
+  private async issueCredential(issueCredentialRequest: CredentialRequest): Promise<CredentialResponse> {
```

Comment:
Just create a callback function that accepts both the CredentialRequest as well as a W3CCredential/ICredential

Reply:
added callback function